File: 2021/day01/Day1.kt
import java.io.File

fun List<String>.ints(): List<Int> = map{ it.toInt() }

open class Day(day: Int, year: Int, debug: Boolean = false) {
    init {
        println("AoC ${year}, day ${day}")
    }

    val inputFile = if (debug) { "sample.txt" } else { "input.txt"}

    val inputList : List<String> = File(inputFile).readLines()

    open fun title(): String = "Day x"

    open fun partOne(): Int {
        return 0
    }

    open fun partTwo(): Int {
        return 0
    }  

    init {
        println(partOne())
        println(partTwo())
    }
}


class Day1(debug: Boolean) : Day(1, 2021, debug) {
    override fun title()= "Sonar Sweep"

    override fun partOne() = solve(1)

    override fun partTwo() = solve(3)

    private fun solve(size: Int): Int {
        return inputList.ints()
            .windowed(size).map{ it.sum() }
            .windowed(2).count{ it.last() > it.first() }
    }
}

fun main(args: Array<String>) {
    Day1(args.size > 0)
}

File: 2019/day01/day1.py
import sys
file = "sample.txt" if len(sys.argv) == 2 and sys.argv[1] == "-d" else "input.txt"
inputs = [int(line) for line in open(file, "r").readlines()]

def ex1():
    return sum(map(lambda x: int(x/3)-2, inputs))

def fuel_cost(mass):
    fuel = int(mass / 3) - 2
    if fuel < 0:
        return 0
    else:
        return fuel + fuel_cost(fuel)

def ex2():
    return sum(map(fuel_cost, inputs))

print("ex1 : %d" % ex1())
print("ex2 : %d" % ex2())